File: sample-rest-api/src/test/java/nl/haploid/octowight/TestData.java
package nl.haploid.octowight;

import nl.haploid.octowight.repository.BookDmo;

import java.util.UUID;

public class TestData {

	public static BookDmo bookDmo(final String genre) {
		final BookDmo book = new BookDmo();
		book.setId(null);
		book.setGenre(genre);
		book.setTitle(title());
		return book;
	}

	public static String title() {
		return UUID.randomUUID().toString();
	}
}

File: event-consumer/src/test/scala/nl/haploid/octowight/consumer/TestData.scala
package nl.haploid.octowight.consumer

import java.util.{Random, UUID}

import nl.haploid.octowight.AtomChangeEvent
import nl.haploid.octowight.registry.data.{Atom, ResourceRoot}

object TestData {

  def atom = new Atom(nextLong, nextString, nextString)

  def atomChangeEvent(atomCategory: String) = AtomChangeEvent(id = nextLong, origin = nextString, category = atomCategory)

  def nextLong: Long = new Random().nextLong

  def nextString = UUID.randomUUID.toString

  def resourceRoot(resourceId: Long) =
    new ResourceRoot(resourceId = resourceId, resourceCollection = nextString, root = atom, version = null)
}
